packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -62,6 +64,18 @@ export function getProxyForField(field: FlexTreeField): TreeNode | TreeValue | u
 			// TODO: Ideally, we would return leaves without first boxing them.  However, this is not
 			//       as simple as calling '.content' since this skips the node and returns the FieldNode's
 			//       inner field.
+
+			const maybeUnboxedContent = asValue.content;
```

Comment:
This is invalid due to the reason documented in the TODO above.

I think a better way to implement this optimization is to first disable unboxing of field nodes OR to stop using field nodes for arrays (I eventually want to delete field nodes from the flex-tree API as they only exist to provide a more ergonomic schema aware API and thats not benefiting us significantly and its causing this issue)

Once thats done, probably as its own PR (and all tests have been updated as needed), then the above TODO can be removed and you can apply the change you have here as an optimization.

Then as another PR, we could look into adjusting the unboxing policy for leaves in flex-tree to make unions including leaf types faster.

Reply:
Discussed with Craig, this wasn't accurate.